### ISSUE TITLE
Build error under Apple M1

### ISSUE THREAD

**Issue by darrenfu** [facebookincubator/velox#3324]
Hi, I'm using an Apple M1 Macbookpro (MacOS 12.6.1).

I'm experiencing the following SIMD related build error, hope someone can take a look and fix it.

From the error stack, the xsimd::batch() constructor only doesn't take two arguments with macro XSIMD_WITH_NEON (Apple M1).
I can workaround the build failure by simply changing to:
```c++
#elif XSIMD_WITH_SSE2
inline xsimd::batch<int64_t> cvtU32toI64(simd::Batch64<int32_t> values) {
  int64_t lo = static_cast<uint32_t>(values.data[0]);
  int64_t hi = static_cast<uint32_t>(values.data[1]);
  return xsimd::batch<int64_t>(lo, hi);
}
#elif XSIMD_WITH_NEON
inline xsimd::batch<int64_t> cvtU32toI64(simd::Batch64<int32_t> values) {
  return xsimd::batch<int64_t>();
}
#endif
```

```
/Users/dofu/git/velox/./velox/dwio/common/ColumnVisitors.h:689:10: error: no matching constructor for initialization of 'xsimd::batch<int64_t>' (aka 'batch<long long>')
  return xsimd::batch<int64_t>(lo, hi);
         ^                     ~~~~~~
/Users/dofu/git/velox/third_party/xsimd/include/xsimd/config/../types/../types/xsimd_batch.hpp:185:9: note: candidate template ignored: could not match 'integer_sequence<unsigned long, _Ip...>' against 'long long'
        batch(T const* data, detail::index_sequence<Is...>) noexcept;
        ^
/Users/dofu/git/velox/third_party/xsimd/include/xsimd/config/../types/../types/xsimd_batch.hpp:47:9: note: candidate constructor not viable: requires single argument 'val', but 2 arguments were provided
        batch(T val) noexcept;
        ^
/Users/dofu/git/velox/third_party/xsimd/include/xsimd/config/../types/../types/xsimd_batch.hpp:48:9: note: candidate constructor not viable: requires single argument 'data', but 2 arguments were provided
        batch(std::initializer_list<T> data) noexcept;
        ^
/Users/dofu/git/velox/third_party/xsimd/include/xsimd/config/../types/../types/xsimd_batch.hpp:49:18: note: candidate constructor not viable: requires single argument 'b', but 2 arguments were provided
        explicit batch(batch_bool_type const& b) noexcept;
                 ^
/Users/dofu/git/velox/third_party/xsimd/include/xsimd/config/../types/../types/xsimd_batch.hpp:50:9: note: candidate constructor not viable: requires single argument 'reg', but 2 arguments were provided
        batch(register_type reg) noexcept;
        ^
/Users/dofu/git/velox/third_party/xsimd/include/xsimd/config/../types/../types/xsimd_batch.hpp:35:11: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 2 were provided
    class batch : public types::simd_register<T, A>
          ^
/Users/dofu/git/velox/third_party/xsimd/include/xsimd/config/../types/../types/xsimd_batch.hpp:35:11: note: candidate constructor (the implicit move constructor) not viable: requires 1 argument, but 2 were provided
/Users/dofu/git/velox/third_party/xsimd/include/xsimd/config/../types/../types/xsimd_batch.hpp:46:9: note: candidate constructor not viable: requires 0 arguments, but 2 were provided
        batch() = default;
        ^
```

**Comment by kgpai** [comment]
@darrenfu Was this issue resolved ?

**Comment by darrenfu** [comment]
Looks like the build issue was resolved when switching to 894493b.
Close the issue for now. Thanks, @Yuhta
